File: python/hex converter/hex.py
#!/usr/bin/env python3

import sys
# import this

if sys.version_info.major == 2:
    print("You are running Python 2, which is no longer supported. Please update to Python 3.")

hex = input("Enter hexadecimal value: ")

print("Here is your flag:")
print(bytes.fromhex(hex).decode("utf-8"))

File: c/hex converter/hex.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    char hex[100];
    printf("Enter hexadecimal value: ");
    scanf("%s", hex);

    // Convert hexadecimal to ASCII
    int len = strlen(hex);
    for (int i = 0; i < len; i += 2) {
        char byte[3] = {hex[i], hex[i + 1], '\0'};
        printf("%c", (char)strtol(byte, NULL, 16));
    }
    printf("\n");

    return 0;
}
